packages/salesforcedx-vscode-core/src/settings/pushOrDeployOnSave.ts

```diff
@@ -86,6 +89,19 @@ export class DeployQueue {
             toDeploy
           );
         }
+
+        telemetryService.sendEventData(
+          'deployOnSave',
+          {
+            deployType: orgType === OrgType.SourceTracked ? 'Push' : 'Deploy'
+          },
+          {
+            documentsToDeploy: toDeploy.length,
+            waitTimeForLastDeploy: this.deployWaitStart
+              ? parseFloat(telemetryService.getEndHRTime(this.deployWaitStart))
```

Comment:
You might want to move the `telemetry.getEndHRTime` changes to a new method in this file for now. One thing to keep in mind is that the rest of the commands returns any time measure in milliseconds. Sending the time info in milliseconds (5400 for 5 seconds and 400milliseconds) makes is easier to build dashboards and queries using the time info since there's no pre-formatting needed.